src/main/java/com/zpi/token/domain/TokenService.java

```diff
@@ -0,0 +1,35 @@
+package com.zpi.token.domain;
+
+import com.zpi.token.api.authorizationRequest.ErrorResponseDTO;
+import com.zpi.token.api.authorizationRequest.RequestDTO;
+import com.zpi.token.api.authorizationRequest.ResponseDTO;
+import com.zpi.token.domain.authorizationRequest.request.InvalidRequestException;
+import com.zpi.token.domain.authorizationRequest.request.RequestValidation;
+import com.zpi.token.domain.authorizationRequest.response.Response;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class TokenService {
+    private final WebClientRepository clientRepository;
+
+    public ResponseEntity<?> validateAuthorizationRequest(RequestDTO requestDTO) {
+        var request = requestDTO.toDomain();
+        var client = clientRepository.getByKey(request.getClientId());
+
+        var validator = new RequestValidation(request, client.orElse(null));
```

Comment:
can we handle request without client?

Reply:
No, lack of client will fail validation step

Reply:
so let's move checking if its empty entirely to validation